src/Domain/Triggers.cpp

```diff
@@ -0,0 +1,81 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/Triggers.hpp"
+
+#include <cstddef>
+#include <memory>
+#include <pup.h>
+#include <string>
+#include <unordered_map>
+
+#include "DataStructures/Tensor/EagerMath/Norms.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/Block.hpp"
+#include "Domain/Domain.hpp"
+#include "Domain/Structure/Element.hpp"
+#include "Domain/Structure/ElementId.hpp"
+
+namespace Triggers {
+SeparationLessThan::SeparationLessThan(const double separation)
+    : separation_(separation) {}
+
+bool SeparationLessThan::operator()(
+    const double time, const ::Domain<3>& domain, const Element<3>& element,
+    const std::unordered_map<
+        std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
+        functions_of_time,
+    const tnsr::I<double, 3, Frame::Grid>& object_center_a,
+    const tnsr::I<double, 3, Frame::Grid>& object_center_b) const {
+  const ElementId<3>& element_id = element.id();
+  const size_t block_id = element_id.block_id();
+
+  const Block<3>& block = domain.blocks()[block_id];
```

Comment:
I'm confused about what's going on here.  It looks like you are using the maps from the current block to map the object centers.  Are these maps guaranteed to be the same for every element?

Reply:
So I wasn't sure how to do this because we literally don't have elements at the centers of the horizons. So I can't check if the center is in an element/block to specifically use that map. But this trigger needs to be run on all elements. My solution was to make use of the fact that currently, all Grid->Distorted maps leave the centers of the objects unchanged, and all Distorted->Inertial maps are global. So if a block didn't have a distorted frame, I just call the Grid->Inertial map because I know they are (currently) defined everywhere. And if a block has a distorted frame, I just call the Distorted->Inertial map which (currently) should be the same as the Grid->Inertial map without a distorted frame.

This is all very hacky. But I couldn't think of a better way. Ideas are welcome

Reply:
OK, it looks like that's true in SpEC, so it should be fine here, at least for now.

Reply:
Yeah while we have Shape/Size/Rotaiton/Expansion/Translation, this will be true. Once we add Skew/Cut-X it won't be and we'll need to find a different way. Possibly we add some kind of thing that holds coordinate maps for inside the excision, except it doesn't actually have grid points/solutions.

But for now, I'll leave it as is.

Reply:
Oh, you're right.  I misread the SpEC paper.  How does SpEC do this?

Reply:
SpEC's maps are all global and then if-else branches depending on coordinate locations inside the calls

Reply:
Ah ok yeah that explains it. I'm thinking that we just have to imbue some object with coordinate maps that are valid inside the excision spheres. Maybe the `ExcisionSphere`s?

Reply:
Yea I think adding coord maps to excision spheres would be reasonable. I think @nikwit needs something like that as well for worldtube things.

Reply:
I agree, that makes sense.  Excision spheres are kind of like blocks in that they are covering a section of the grid-coordinates, just not a section with elements, so having some similar properties is reasonable.

Reply:
Ok I'll add maps to the `ExcisionSphere`s (probably in a separate PR).

Reply:
Sounds good.  I don't think we need to hold this PR up for it.  Just fix it in the PR that adds the feature.  If you're not going to do it immediately file a bug so we don't forget.